quipucords/scanner/openshift/api.py

```diff
@@ -291,3 +309,33 @@ def _init_cluster(self, cluster) -> OCPCluster:
             version=cluster["status"]["desired"]["version"],
         )
         return ocp_cluster
+
+    def _init_managed_cluster(self, cluster) -> dict:
+        managed_cluster_metrics = {}
+
+        key_map_labels = {
+            "vendor": "vendor",
+            "cloud": "cloud",
+            "version": "openshiftVersion",
+            "managed_cluster_id": "clusterID",
+        }
+
+        labels = cluster["metadata"]["labels"]
+        for metric_key, cluster_api_key in key_map_labels.items():
+            managed_cluster_metrics[metric_key] = labels[cluster_api_key]
+
+        conditions = cluster["status"]["conditions"]
+        is_cluster_available = any(
+            condition.get("reason") == "ManagedClusterAvailable"
+            for condition in conditions
+        )
+        managed_cluster_metrics["available"] = is_cluster_available
+
+        capacity = cluster["status"]["capacity"]
+        if capacity:
+            managed_cluster_metrics["core_worker"] = capacity.get("core_worker", "")
+            managed_cluster_metrics["socket_worker"] = capacity.get("socket_worker", "")
```

Comment:
could core_worker/socket_worker not be available when capacity is present? and do you know why capacity might not be present?

Reply:
Capacity might not be present because we are looking at things from the hub perspective. So this endpoint will retrieve all the managed clusters, regardless their state: clusters that were connected in the past but now are disconnected, active clusters, and clusters that you wanted to import but had any sort of failure.If you were not able to fully connect to a cluster, then you couldn't collect detailed info about it, this is when conditions is not available. 
I haven't seen any case when capacity is present but not info on socket_workers and core_workers.

Looking at the code from the repo that generates those metrics. If capacity is present, then it will return some sort of value for core_worker and socket_worker.

https://github.com/stolostron/clusterlifecycle-state-metrics/blob/495ed8be90cf4a27ae9fac0c2518320462c1e46a/pkg/generators/cluster/managedclusterinfo.go#L160